src/main/scala/com/typesafe/sbt/packager/jar/ClasspathJarPlugin.scala

```diff
@@ -1,46 +1,54 @@
 package com.typesafe.sbt.packager
 package archetypes.jar
 
+import java.io.File
+import java.util.jar.Attributes
+
+import sbt.Package.ManifestAttributes
 import sbt._
-import sbt.Keys.{ mappings, target, name, mainClass, sourceDirectory }
-import com.typesafe.sbt.packager.Keys.{ scriptClasspath }
-import com.typesafe.sbt.SbtNativePackager.{ Universal }
+import sbt.Keys._
+import com.typesafe.sbt.packager.Keys._
+import com.typesafe.sbt.SbtNativePackager.Universal
 
 object ClasspathJarPlugin extends AutoPlugin {
 
   object autoImport {
-    val classspathJarName = TaskKey[String]("classpath-jar-name", "classpath-jar name")
+    val packageJavaClasspathJar = TaskKey[File]("packageJavaClasspathJar", "Creates a Java classpath jar that specifies the classpath in its manifest")
   }
   import autoImport._
 
   override def requires = archetypes.JavaAppPackaging
 
-  override lazy val projectSettings = Seq[Setting[_]](
-    classspathJarName <<= (Keys.packageBin in Compile, Keys.projectID, Keys.artifact in Compile in Keys.packageBin) map { (jar, id, art) =>
-      makeJarName(id.organization, id.name, id.revision, art.name, art.classifier)
+  override lazy val projectSettings = Defaults.packageTaskSettings(packageJavaClasspathJar, mappings in packageJavaClasspathJar) ++ Seq(
+    mappings in packageJavaClasspathJar := Nil,
+
+    artifactClassifier in packageJavaClasspathJar := Option("classpath"),
+
+    packageOptions in packageJavaClasspathJar := {
+      val classpath = (scriptClasspath in packageJavaClasspathJar).value
```

Comment:
This will pick up the normal `scriptClasspath` setting, meanwhile we override the one that goes into `bashScriptDefines` and `batScriptReplacements` below to be nothing.